tests/integration/service/test_working_schema_service.py

```diff
@@ -73,27 +72,30 @@ def migration_service_fixture(user, persistent, get_cred):
     return service
 
 
-@pytest_asyncio.fixture
-async def patch_list_tables_to_exclude_datasets(session, dataset_registration_helper):
+@pytest.fixture
+def patch_to_exclude_datasets(dataset_registration_helper):
     """
-    Fixture to patch list_tables to exclude dataset tables
+    Fixture to patch remove_materialized_tables to exclude dataset tables
 
     The dataset tables are stored in the metadata schema and will be dropped during the schema
     recreation process, but these datasets themselves are required to complete the process (when
     re-online-enabling features)
     """
-    assert session.source_type == "snowflake"
-
-    async def patched_list_tables(database_name=None, schema_name=None):
-        tables = (
-            await session.execute_query(f'SHOW TABLES IN SCHEMA "{database_name}"."{schema_name}"')
-        )["name"].tolist()
-        known_tables = set(dataset_registration_helper.table_names)
-        tables = [t for t in tables if t not in known_tables]
-        return tables
-
-    with patch("featurebyte.session.snowflake.SnowflakeSession.list_tables") as p:
-        p.side_effect = patched_list_tables
+    from featurebyte.session.base import BaseSchemaInitializer
+
+    original_func = BaseSchemaInitializer.remove_materialized_tables
+
+    def patched_remove_materialized_tables(table_names):
```

Comment:
Wonder if it would be better register source tables in a separate schema in tests. Since the helper is a session fixture we can add a check during teardown to make sure the tables are intact

Reply:
Sure, will refactor in a next PR. I think that was the initial approach, but many tests were broken because they relied on the assumption that the test data are in the same schema. But I agree that it is more proper to fix those tests.